src/groupFrom/types.ts

```diff
@@ -1,10 +1,48 @@
-import {UploadcareGroupInterface} from '../types'
-import {UploadInterface} from '../lifecycle/types'
-import {HandlerInterface} from '../fileFrom/types'
+import {FileData} from '../types'
+import {Url, Uuid} from '..'
+import {isFileData, isUrl, isUuid} from '../fileFrom/types'
 
 /**
- * Base `thenable` interface for uploading `groupFrom` (`object`, `url`, `input`, `uploaded`).
+ * FileData type guard.
+ *
+ * @param {FileData | Url | Uuid} data
  */
-export interface GroupUploadInterface extends UploadInterface<UploadcareGroupInterface> {}
+export const isFileDataArray = (data: FileData[] | Url[] | Uuid[]): data is FileData[] => {
+  for (const item of data) {
```

Comment:
Probably we can use `.every`
It is supported by every modern browsers, but if we will use upload-client in the current widget, it can drop compatibility with the very old browsers %(